common/src/main/java/me/lucko/luckperms/common/storage/implementation/sql/connection/hikari/MariaDbConnectionFactory.java

```diff
@@ -50,23 +50,10 @@ protected String defaultPort() {
 
     @Override
     protected void configureDatabase(HikariConfig config, String address, String port, String databaseName, String username, String password) {
-        config.setDataSourceClassName("org.mariadb.jdbc.MariaDbDataSource");
```

Comment:
Is there a reason for switching to JDBC URL instead of the DataSource?

Reply:
> Is there a reason for switching to JDBC URL instead of the DataSource?

idk how to set databaseName with `MariaDbDataSource` in 3.x version

Reply:
@lucko

Reply:
Should just be a case of finding the datasource class and defining it as we did previously.

It will implement the DataSource interface

Reply:
> Should just be a case of finding the datasource class and defining it as we did previously.
> 
> It will implement the DataSource interface

```java
config.addDataSourceProperty("databaseName", databaseName);
```

will throw error like "databaseName not found"

also, MariaDbDataSource from 2.x and MariaDbDataSource from 3.x are very different
![image](https://user-images.githubusercontent.com/54859825/226463272-5b3b5fb5-5fe5-414c-b3e5-8d1e09d5a6bb.png)
and in 3.x there is no setProperties method, in 2.x it is deprecated


hmm, maybe just replace setDriverClassName with setDataSourceClassName?

Reply:
@lucko